clearnode/rpc_router_private.go

```diff
@@ -199,13 +204,15 @@ func (r *RPCRouter) HandleTransfer(c *RPCContext) {
 	}
 
 	// Allow only ledger accounts as destination at the current stage. In the future we'll unlock application accounts.
-	if params.Destination == "" || params.Destination == c.UserID || !common.IsHexAddress(params.Destination) {
+	switch {
+	case params.Destination == "" && params.DestinationTag == "":
+		c.Fail("either destination or destination_tag must be provided")
```

Comment:
The error message is slightly misleading. We know that both could be provided. It boils down to the word `either`; instead of using the `aut aut` form (i.e. XOR), we should use the `vel` one (i.e. OR):

```suggestion
		c.Fail("destination or destination_tag must be provided")
```